src/satosa/backends/idpy_oidc.py

```diff
@@ -0,0 +1,144 @@
+
+        :type context: satosa.context.Context
+        :type internal_request: satosa.internal.InternalData
+        :rtype satosa.response.Redirect
+        """
+        login_url = self.client.init_authorization()
+        return Redirect(login_url)
+
+    def register_endpoints(self):
+        """
+        Creates a list of all the endpoints this backend module needs to listen to. In this case
+        it's the authentication response from the underlying OP that is redirected from the OP to
+        the proxy.
+        :rtype: Sequence[(str, Callable[[satosa.context.Context], satosa.response.Response]]
+        :return: A list that can be used to map the request to SATOSA to this endpoint.
+        """
+        url_map = []
+        redirect_path = self.client.context.claims.get_usage('redirect_uris')
+        if not redirect_path:
+            raise SATOSAError("Missing path in redirect uri")
+        redirect_path = urlparse(redirect_path[0]).path
+        url_map.append((f"^{redirect_path.lstrip('/')}$", self.response_endpoint))
+        return url_map
+
+    def response_endpoint(self, context, *args):
+        """
+        Handles the authentication response from the OP.
+        :type context: satosa.context.Context
+        :type args: Any
+        :rtype: satosa.response.Response
+
+        :param context: SATOSA context
+        :param args: None
+        :return:
+        """
+
+        _info = self.client.finalize(context.request)
+        self._check_error_response(_info, context)
+        userinfo = _info.get('userinfo')
+        id_token = _info.get('id_token')
+
+        if not id_token and not userinfo:
+            msg = "No id_token or userinfo, nothing to do.."
+            logline = lu.LOG_FMT.format(id=lu.get_session_id(context.state), message=msg)
+            logger.error(logline)
+            raise SATOSAAuthenticationError(context.state, "No user info available.")
+
+        all_user_claims = dict(list(userinfo.items()) + list(id_token.items()))
+        msg = "UserInfo: {}".format(all_user_claims)
+        logline = lu.LOG_FMT.format(id=lu.get_session_id(context.state), message=msg)
+        logger.debug(logline)
+        internal_resp = self._translate_response(all_user_claims, _info["issuer"])
+        return self.auth_callback_func(context, internal_resp)
+
+    def _translate_response(self, response, issuer):
+        """
+        Translates oidc response to SATOSA internal response.
+        :type response: dict[str, str]
+        :type issuer: str
+        :type subject_type: str
+        :rtype: InternalData
+
+        :param response: Dictioary with attribute name as key.
+        :param issuer: The oidc op that gave the repsonse.
+        :param subject_type: public or pairwise according to oidc standard.
+        :return: A SATOSA internal response.
+        """
+        auth_info = AuthenticationInformation(UNSPECIFIED, str(datetime.now()), issuer)
```

Comment:
We used to use `datetime.now` in the old module, but I think we should set the the `auth_time` of the id-token. 

Similarly, for the ACCR we should set `acr` or `amr` (not sure which of the two fits better) if present.

```suggestion
        auth_class_ref = response["auth_time"]
        timestamp = response.get("amr", UNSPECIFIED)
        auth_info = AuthenticationInformation(auth_class_ref, timestamp, issuer)
```

Reply:
The attribute names confuses me completely.

`auth_time` in the id-token has nothing to do with `acr` or `amp` so I don't understand this change.

Shouldn't it be something like:

```
timestamp = response["auth_time"]
auth_class_ref = response.get("amr", response.get("acr", UNPECIFIED))
auth_info = AuthenticationInformation(auth_class_ref, timestamp, issuer)
```

Reply:
Or perhaps 
`auth_class_ref = response.get("acr", response.get("amr", UNPECIFIED))`
is a better order